File: CSharp/ImageProcessing/Views/FormHistgramLiveCharts.cs
using LiveCharts;
using LiveCharts.Wpf;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Drawing.Imaging;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace ImageProcessing
{
    /// <summary>
    /// FormHistgramLiveCharts のロジック
    /// </summary>
    public partial class FormHistgramLiveCharts : Form
    {
        private readonly ComHistgramLiveCharts mHistgramChart;

        /// <summary>
        /// オリジナルのビットマップ
        /// </summary>
        public Bitmap BitmapOrg
        {
            set { mHistgramChart.BitmapOrg = value; }
            get { return mHistgramChart.BitmapOrg; }
        }

        /// <summary>
        /// 画像処理後のビットマップ
        /// </summary>
        public Bitmap BitmapAfter
        {
            set { mHistgramChart.BitmapAfter = value; }
            get { return mHistgramChart.BitmapAfter; }
        }

        /// <summary>
        /// Formのオープン状態
        /// </summary>
        public bool IsOpen { set; get; }

        /// <summary>
        /// コンストラクタ
        /// </summary>
        public FormHistgramLiveCharts()
        {
            InitializeComponent();

            mHistgramChart = new ComHistgramLiveCharts();
        }

        /// <summary>
        /// グラフの描画
        /// </summary>
        public void DrawHistgram()
        {
            var items = mHistgramChart.DrawHistgram();

            chart.Series.Clear();
            for (int index = 0; index < items.Count; index++)
            {
                chart.Series.Add(items[index]);
            }

            return;
        }

        /// <summary>
        /// Formのクローズ処理
        /// </summary>
        /// <param name="sender">オブジェクト</param>
        /// <param name="e">FormClosingイベントのデータ</param>
        private void OnFormClosingFormHistgramLiveCharts(object sender, FormClosingEventArgs e)
        {
            IsOpen = false;

            return;
        }

        /// <summary>
        /// メニューのクリック
        /// </summary>
        /// <param name="sender">オブジェクト</param>
        /// <param name="e">イベントのデータ</param>
        public void OnClickMenu(object sender, EventArgs e)
        {
            string menuTitle = sender.ToString();

            switch (menuTitle)
            {
                case ComInfo.MENU_SAVE_CSV:
                    SaveCsv();
                    break;
                default:
                    break;
            }

            return;
        }

        /// <summary>
        /// CSVファイル保存
        /// </summary>
        public void SaveCsv()
        {
            if (!mHistgramChart.SaveCsv())
            {
                MessageBox.Show(this, "Save CSV File Error", "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
            }

            return;
        }
    }
}

File: Vb/ImageProcessing/Views/FormHistgramLiveCharts.vb
﻿Imports LiveCharts
Imports LiveCharts.Wpf

''' <summary>
''' FormHistgramLiveCharts のロジック
''' </summary>
Public Class FormHistgramLiveCharts
    Private mHistgramChart As ComHistgramLiveCharts

    ''' <summary>
    ''' オリジナルのビットマップ
    ''' </summary>
    Public Property BitmapOrg() As Bitmap
        Set(value As Bitmap)
            mHistgramChart.BitmapOrg = value
        End Set
        Get
            Return mHistgramChart.BitmapOrg
        End Get
    End Property

    ''' <summary>
    ''' 画像処理後のビットマップ
    ''' </summary>
    Public Property BitmapAfter() As Bitmap
        Set(value As Bitmap)
            mHistgramChart.BitmapAfter = value
        End Set
        Get
            Return mHistgramChart.BitmapOrg
        End Get
    End Property

    ''' <summary>
    ''' Formのオープン状態
    ''' </summary>
    Public Property IsOpen() As Boolean

    ''' <summary>
    ''' コンストラクタ
    ''' </summary>
    Public Sub New()

        ' この呼び出しはデザイナーで必要です。
        InitializeComponent()

        ' InitializeComponent() 呼び出しの後で初期化を追加します。

        mHistgramChart = New ComHistgramLiveCharts()
    End Sub

    ''' <summary>
    ''' グラフの描画
    ''' </summary>
    Public Sub DrawHistgram()
        Dim items = New List(Of LineSeries)
        items = mHistgramChart.DrawHistgram()

        chart.Series.Clear()
        For index As Integer = 0 To items.Count - 1
            chart.Series.Add(items(index))
        Next

        Return
    End Sub

    ''' <summary>
    ''' Formのクローズ処理
    ''' </summary>
    ''' <param name="sender">オブジェクト</param>
    ''' <param name="e">FormClosingイベントのデータ</param>
    Private Sub OnFormClosingFormHistgramLiveCharts(sender As Object, e As FormClosingEventArgs) Handles MyBase.FormClosing
        IsOpen = False

        Return
    End Sub

    ''' <summary>
    ''' メニューのクリック
    ''' </summary>
    ''' <param name="sender">オブジェクト</param>
    ''' <param name="e">イベントのデータ</param>
    Private Sub OnClickMenu(sender As Object, e As EventArgs) Handles SaveCsvToolStripMenuItem.Click
        Dim menuTitle = sender.ToString()

        Select Case (menuTitle)
            Case ComInfo.MENU_SAVE_FILE
                SaveCsv()
            Case Else
        End Select
    End Sub

    ''' <summary>
    ''' CSVファイル保存
    ''' </summary>
    Public Sub SaveCsv()
        If (mHistgramChart.SaveCsv() = False) Then
            MessageBox.Show(Me, "Save CSV File Error", "Error", MessageBoxButtons.OK, MessageBoxIcon.Error)
        End If
    End Sub
End Class